File: users-client/src/main/java/net/plshark/auth/throttle/UsernameExtractor.java
package net.plshark.auth.throttle;

import java.util.Optional;

import org.springframework.http.server.reactive.ServerHttpRequest;

/**
 * Extracts usernames from requests
 */
public interface UsernameExtractor {

    /**
     * Extract the username from a request
     * @param request the request
     * @return the username or empty if not found
     */
    Optional<String> extractUsername(ServerHttpRequest request);
}

File: users-client/src/main/kotlin/net/plshark/users/auth/throttle/UsernameExtractor.kt
package net.plshark.users.auth.throttle

import org.springframework.http.server.reactive.ServerHttpRequest
import java.util.*

/**
 * Extracts usernames from requests
 */
interface UsernameExtractor {
    /**
     * Extract the username from a request
     * @param request the request
     * @return the username or empty if not found
     */
    fun extractUsername(request: ServerHttpRequest): Optional<String>
}